config.json

```diff
@@ -1310,6 +1310,18 @@
           "custom-types"
         ],
         "difficulty": 6
+      },
+      {
+        "slug": "knapsack",
+        "name": "Knapsack",
+        "uuid": "9bf186d3-e904-42f2-a705-dffbe98adb08",
+        "practices": [],
+        "prerequisites": [
+          "comparison",
+          "lists",
+          "records"
+        ],
+        "difficulty": 6
```

Comment:
Feel like a 7 or an 8, because the algorithm isn't trivial. What do you think?

Reply:
I think a 7 is also reasonable. My hesitance with going to an 8 is seeing Zebra puzzle at 8 (and me thinking its not quite as hard Zebra Puzzle). The other thought I have is this example solution uses memoization. A simpler solution could be to use recursion without the memoization (although it won't be as efficient).

Reply:
Yes you're right, it's not quite a Zebra puzzle.
In the end it's not an exact science, but 7 feels right :)